src/core_plugins/kibana/public/dashboard/dashboard_state_manager.js

```diff
@@ -138,20 +138,23 @@ export class DashboardStateManager {
   }
 
   _handleStoreChanges() {
-    if (this._areStoreAndAppStatePanelsEqual()) {
-      return;
+    let dirty = false;
+    if (!this._areStoreAndAppStatePanelsEqual()) {
+      const state = store.getState();
+      this.appState.panels = [];
+      _.map(getPanels(state), panel => {
```

Comment:
If we're just pushing the panels from one array to another, then I think we can do this to be a little clearer:

```js
this.appState.panels = [].concat(panels);
```

Reply:
If `panels` is null, it won't work though, but you could do `this.appState.panels = [...panels || [] ];`

Reply:
I don't think either of those will work because panels on the store is an object mapping of id to panel, not an array like it is in appState..

Reply:
> I don't think either of those will work because panels on the store is an object mapping of id to panel, not an array like it is in appState..

So `_.map` operates on an Object returned by `getPanels()` and just uses the values?

Reply:
yup!  Now that you put it like that, I realize I can switch to Object.values instead of lodash.  Maybe that will help clarify the difference